fy22_ferry_study/attraction_travel_times/distance_direction_parser.py

```diff
@@ -0,0 +1,164 @@
+import googlemaps
+from datetime import datetime
+import os
+from dotenv import load_dotenv
+from pathlib import Path
+import pandas as pd
+from shapely.geometry import MultiLineString
+import polyline
+import geopandas as gpd
+
+load_dotenv()
+api_key = os.getenv("google_api")
+gmaps = googlemaps.Client(key=api_key)
+GDRIVE_FOLDER = Path(os.getenv("GDRIVE_FOLDER"))
+origins = GDRIVE_FOLDER / "origin.csv"
+destinations = GDRIVE_FOLDER / "attractions.csv"
+origin_df = pd.read_csv(origins)
+destinations_df = pd.read_csv(destinations)
+now = datetime.now()
+
+
+origin = (
+    39.946196,
+    -75.139832,
+)  # hardcoded here since my origin doesn't change, but if yours does you can tweak the function to accept origins.
+
+
+def distance_duration_iteration(mode):
+    """returns a list of origin/destination dicts for driving or transit, respectively"""
+    driving_list = []
+    transit_list = []
+
+    def distance_duration(destination):
+        output = gmaps.directions(
+            origin,
+            destination,
+            mode,
+            units="imperial",
+            departure_time=now,
+        )
+        names = {"name": name}
+        output.insert(0, names)
+
+        if mode == "driving":
+            driving_list.append(output)
+
+        if mode == "transit":
+            transit_list.append(output)
+
+    """loops through the destinations file and runs the distance duration function"""
+    for idx, row in destinations_df.iterrows():
+        lat = row["Latitude"]
+        lon = row["Longitude"]
+        name = row["Name"]
+        destination = (lat, lon)
+        distance_duration(destination)
+
+    if mode == "driving":
+        print("returning driving list")
+        return driving_list
+    if mode == "transit":
+        print("returning transit list")
+        return transit_list
+
+
+def save_list_in_memory(transpo_list):
```

Comment:
I don't think this function does what the docstring says it does. The input arg should already be in-memory.

Reply:
Yeah, I think this is redundant. I was initially using it in a Jupyter notebook just as a way to hold the value without calling the API every time I ran the distanceduration() function but I realize that I could just use a variable and achieve the same effect